src/components/SiteTrackingRequired/SiteTrackingRequired.js

```diff
@@ -20,18 +21,23 @@ export const SiteTrackingRequired = InjectAppServices(
     dependencies: {
       appConfiguration: { dopplerLegacyUrl },
       dopplerLegacyClient,
-      sessionManager,
     },
   }) => {
-    const [isActivatingTrial, setIsActivatingTrial] = useState(false);
+    const [state, setState] = useState({ isLoading: false });
+
+    if (state.isActivatedTrial) {
+      return <RedirectToLegacyUrl to="/ControlPanel/CampaignsPreferences/SiteTrackingSettings" />;
+    }
 
     const activateTrial = async () => {
-      setIsActivatingTrial(true);
+      setState({ isLoading: true });
       try {
-        await dopplerLegacyClient.activateSiteTrackingTrial();
-        sessionManager.restart();
+        const isActivatedTrial = await dopplerLegacyClient.activateSiteTrackingTrial();
```

Comment:
I think that we should not pay attention to the result of `activateSiteTrackingTrial`, when it fails it throws an exception:

https://github.com/FromDoppler/doppler-webapp/blob/f859f0285f78073bbc2d341c891f380e4ca11000/src/services/doppler-legacy-client.ts#L420-L428

```suggestion
        await dopplerLegacyClient.activateSiteTrackingTrial();
```

Reply:
if throw excepcion dont need update the state

Reply:
Right, for that reason I am proposing not paying attention to the result value.

Reply:
Oh, sorry, my bad!, I did not see your changes in `activateSiteTrackingTrial`!

We are using now the same approach of _error-free_ services in this function too, nice!